crates/number/src/nonzero.rs

```diff
@@ -1,20 +1,36 @@
 use {
-    anyhow::anyhow,
+    anyhow::Context,
     primitive_types::U256 as ZeroU256,
     serde::{Deserialize, Deserializer, Serialize, Serializer},
+    std::fmt::{self, Display, Formatter},
 };
 
 #[derive(Copy, Clone, Debug, Hash, Eq, PartialEq, Ord, PartialOrd)]
 pub struct U256(ZeroU256);
+
+impl U256 {
+    pub fn new(value: ZeroU256) -> Option<Self> {
+        if !value.is_zero() {
+            Some(Self(value))
+        } else {
+            None
+        }
```

Comment:
I just have to point out this hidden gem every time I see it. :sweat_smile:
```suggestion
        !value.is_zero().then_some(Self(value))
```

Reply:
I always forget...